agent/api/task/task.go

```diff
@@ -1502,6 +1502,37 @@ func (task *Task) getASMAuthResource() ([]taskresource.TaskResource, bool) {
 	return res, ok
 }
 
+// PopulateSSMSecrets appends the container's env var map with ssm parameters
+func (task *Task) PopulateSSMSecrets(container *apicontainer.Container) *apierrors.DockerClientConfigError {
+	resource, ok := task.getSSMSecretsResource()
+	if !ok {
+		return &apierrors.DockerClientConfigError{"task secret data: unable to fetch SSM Secrets resource"}
```

Comment:
why is this a `DockerClientConfigError`? This error happens when resource is not found in task resources map yeah?

Reply:
yea i was wondering about this. thinking was `PopulateSSMSecrets` is acting on the docker config for this container. failure here halt the start container workflow because the config couldn't be constructed.